Lib/test/test_sys.py

```diff
@@ -1080,7 +1080,14 @@ def test_getallocatedblocks(self):
             # While we could imagine a Python session where the number of
             # multiple buffer objects would exceed the sharing of references,
             # it is unlikely to happen in a normal test run.
-            self.assertLess(a, sys.gettotalrefcount())
+            #
+            # In free-threaded builds each code object owns an array of
+            # pointers to copies of the bytecode. When the number of
+            # code objects is a large fraction of the total number of
+            # references, this can cause the total number of allocated
+            # blocks to exceed the total number of references.
+            if not support.Py_GIL_DISABLED:
```

Comment:
Now that we can free the unused tlbcs, can we replace this with `sys._clear_internal_caches()`

Reply:
Unfortunately, no. It seems to be very sensitive to which kinds of objects are on the heap as well as the number of non reference counted allocations (blocks) per object. With the introduction of TLBC, there is at least one additional block allocated per code object that is not reference counted even if we clear unused TLBCs, the _PyCodeArray. That is enough to trigger the assertion.

This assertion feels pretty brittle and I'd be in favor of removing it, but that's probably worth doing in a separate PR.

Reply:
Maybe replace it with a more meaningful test rather than remove it. But in another PR.